File: 102. Binary Tree Level Order Traversal/102. Binary Tree Level Order Traversal.cpp
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ans;
        queue<TreeNode*> q;
        if(root)    q.push(root);
        while(!q.empty()){
            vector<int> currLevel(q.size(), 0);
            for(auto& num: currLevel){
                TreeNode* curr = q.front();
                num = curr->val;
                if(curr->left)  q.push(curr->left);
                if(curr->right)  q.push(curr->right);
                q.pop();
            }
            ans.push_back(currLevel);
        }
        return ans;
    }
};

File: 102. Binary Tree Level Order Traversal/102. Binary Tree Level Order Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> ans = new ArrayList<List<Integer>>();
        if(root == null)    return ans;
        Queue<TreeNode> q = new LinkedList<TreeNode>();
        q.offer(root);
        while(!q.isEmpty()){
            List level = new ArrayList<>();
            for(int i = q.size(); i > 0; --i){
                TreeNode curr = q.peek();
                if(curr.left != null){
                    q.offer(curr.left);
                }
                if(curr.right != null){
                    q.offer(curr.right);
                }
                level.add(curr.val);
                q.poll();
            }
            ans.add(level);
        }
        return ans;
    }
}